client/src/api/v3/boost.ts

```diff
@@ -1,28 +1,121 @@
 import { emit } from 'api/common/socket';
-import { BoostConfigType } from 'types/boost';
+import {
+  FileConfigT,
+  ConfigBundleT,
+  ConfigObjT,
+  ConfigT,
+  fileConfigTypeToRuntype,
+} from 'types/boost';
+import toast from 'react-hot-toast';
+import { Runtype } from 'runtypes';
+
+type payloadAction = 'upload' | 'delete';
+
+/**
+ * Check that the given content is of the type given
+ *
+ * @param content the content to check for it's type
+ * @param typeRequired the type the content is expected to be
+ *
+ * @returns true if the content satisfies the given type else false
+ */
+function isCorrectContentType(content: any, typeRequired: Runtype) {
+  try {
+    if (typeRequired.check(content)) {
+      return true;
+    }
+  } catch (error) {
+    return false;
+  }
+  return false;
+}
 
 /**
- * Send the content of the given configuration file on `boost/configs/action` over MQTT
+ * Send configuration status over MQTT on topic 'boost/configs/action'
  *
+ * @param actionType represents whether the config is being uploaded or deleted
  * @param type the type of the configuration being sent
- * @param configFile file containing the configuration content
+ * @param name name of the config file
+ * @param configContent configuration content
+ */
+function sendConfig(
+  actionType: payloadAction,
+  type: ConfigT,
+  name: string,
+  configContent: string | null,
+) {
+  const channel = 'send-config';
+  const payload = {
+    action: actionType,
+    configType: type,
+    fileName: name,
+    content: configContent,
+  };
+  // To check things work without needing the server started
+  console.log(payload);
+  emit(channel, JSON.stringify(payload));
+}
+
+const configTypeToFileSuffix: { [K in ConfigT]: string } = {
+  rider: '_rider.json',
+  bike: '_bike.json',
+  track: '_track.json',
+  powerPlan: '_powerPlan.json',
+};
+
+/**
+ * Split the given configurations object into inidvidual configs and then send them over MQTT
+ *
+ * @param configs dictionary containign the 4 config types (`powerPlan`, `rider`, `track` and `bike`)
+ * @param fileName name of the file that contained all the configs
+ */
+function uploadMultipleConfigs(configs: ConfigBundleT, fileName: string) {
+  // For each config, send the config content over MQTT
+  Object.entries(configs).forEach((configEntry) => {
+    const configType = configEntry[0] as ConfigT;
+    const config = ConfigObjT.check(configEntry[1]);
+
+    // Since this config was uploaded as a bundle give it a different file name to differnetiate it's config type.
```

Comment:
typos: Since this config was uploaded as a bundle, give it a different file name to differentiate its config type.